DataAccessLayer/Crud/Implementation/CategoryCrud.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DataAccessLayer.DataBaseModels;
+using DataAccessLayer.Crud.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccessLayer.Crud.Implementation
+{
+    public class CategoryCrud : ICrudDictionary<Category>
```

Comment:
It looks that you are still using "ICrudRepository" approach.
But this time in next layers you use direct classes, not interface, so you do not have this bottleneck now.

But this time please think what benefits does using of "ICrudDictionary" give to you?
You're are not using this interface in next layers.
What is the purpose of using it?